### DISCUSSION TITLE
Deploying to a subdomain

### DISCUSSION THREAD

**Discussion by ghost** [pocketbase/pocketbase#457] [Q&A]
It would be useful to have a subdomain deployment in the documentation as the basic NGINX code assumes there is a dedicated server for pocketbase. I've been facing horrible issues trying to adapt that template correctly.

Let's suppose a user has `domain.com` and it has a static html page there.

What if, on the same server the same user wants to host pocketbase under `api.domain.com` ?

I would assume this could be done using this:

```nginx
server {
    listen 80;
    server_name api.domain.com;

    location / {
        proxy_set_header Connection '';
        proxy_http_version 1.1;
        proxy_read_timeout 180s;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        proxy_redirect off;
        proxy_pass http://127.0.0.1:8090;
    }
}
```

However, this block makes nginx redirect the user to the main `domain.com` content, even tough  the `api.domain.com` URL is being used.

**Comment by jonafeucht** [comment]
Works perfectly fine on my end, on a subdomain.

Here is my config:
```nginx

location ~ / {
	proxy_set_header Connection '';
	proxy_read_timeout 180s;
	proxy_set_header Host $host;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;

	proxy_pass http://localhost:8090;
}
```

**Reply by ghost** [reply]
I had to do extra steps for that :thinking: since I was also using a SSL cert. Does  the `~` in your main path make any difference?

PS: Thank you for your response :)

**Reply by ganigeorgiev** [reply]
@VentGrey Do you have a redirect from 80 -> 443 somewhere? Please provide your full nginx configuration file (including the block where you are loading the certificate).

**Reply by ghost** [reply]
I had to do the following steps:

1- Serve pocketbase to http (exclude `--https` flag) on my systemd service.
2- Use the normal NGINX template provided in the documentation.
3- Add a certbot certificate

In the end I had these two files:

## Systemd Unit File

```ini
[Unit]
Description = Pocketbase server

[Service]
Type = simple
User = nginx
Group = nginx
# LimitNOFILE = 4096
Restart = always
RestartSec = 5s
ExecStart = /etc/extbin/pb/pocketbase serve --http="0.0.0.0:8090"

[Install] 
WantedBy = multi-user.target
```

### NGINX Virtual Host file

```nginx
server {
    server_name api.domain.com;

    root /etc/extbin/pb/;

    location / {
        proxy_set_header Connection '';
        proxy_http_version 1.1;
        proxy_read_timeout 180s;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header X-NginX-Proxy true;

        proxy_pass http://localhost:8090;
    }

    listen [::]:443 ssl; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/api.domain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/api.domain.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = api.domain.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    listen 80;
    listen [::]:80;
    server_name api.domain.com;
    return 404; # managed by Certbot
}
```

**Comment by ghost** [comment]
Update: Even after posting my update I see that this error only happens in EL-Linux distros like Fedora, RHEL, CentOS (not the AUR updated one), Rocky Linux and Alma Linux.

I don't know if further steps are needed to fix those distribution's broken-by-design workflow. It does work on ~.deb based distributions :s so there must be something strange.
